service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -44,22 +44,10 @@
      */
     public abstract List<String> uris();
 
-    // hides implementation details
     public static Builder builder() {
-        return ImmutableServiceConfiguration.builder();
+        return new Builder();
     }
 
-    // hides implementation details
-    public interface Builder {
-
-        Builder apiToken(BearerToken apiToken);
-
-        Builder security(SslConfiguration security);
-
-        Builder uris(Iterable<String> uris);
-
-        Builder from(ServiceConfiguration otherConfig);
-
-        ServiceConfiguration build();
+    public static final class Builder extends ImmutableServiceConfiguration.Builder {
```

Comment:
@nmiyake we are still experimenting with the new pattern, will send out an email to discuss this more in depth. 
@chrisalice all immutable classes are value classes, so builders can be used instead of mocking them in tests which isn't a huge compromise i think :)